### DISCUSSION TITLE
Warning Messages When I Run Webui.bat + Question about Xformers

### DISCUSSION THREAD

**Discussion by gohan2091** [vladmandic/sdnext#1555] [Q&A] [answered]
Hi all,

I'm new to A1111 / Vlads and recently acquired an NVIDIA RTX 4090. I've been exploring its capabilities, and I have a couple of questions:

1) Regarding Xformers, should I enable them? are there any drawbacks like reduced quality or issues with batch modes? or should I be enabling something else instead?

2) Whenever I run A1111, I receive three warning messages that I've been ignoring. How docan I update these things? Is it necessary? Is the log indicating that it is attempting to update them automatically? When I close A1111 and reopen, the same warnings appear. Any ideas?

> 14:25:57-515713 INFO     Starting SD.Next
14:25:57-518715 INFO     Python 3.10.11 on Windows
14:25:57-548741 INFO     Version: 875d0db1 Sun Jul 2 08:26:51 2023 -0400
14:25:57-926059 INFO     nVidia CUDA toolkit detected
14:25:59-074951 INFO     Torch 2.0.1+cu118
14:25:59-088704 INFO     Torch backend: nVidia CUDA 11.8 cuDNN 8700
14:25:59-090706 INFO     Torch detected GPU: NVIDIA GeForce RTX 4090 VRAM 24564 Arch (8, 9) Cores 128
14:25:59-230836 INFO     Enabled extensions-builtin: ['a1111-sd-webui-lycoris', 'clip-interrogator-ext', 'LDSR',
                         'Lora', 'multidiffusion-upscaler-for-automatic1111', 'ScuNET', 'sd-dynamic-thresholding',
                         'sd-extension-system-info', 'sd-webui-agent-scheduler', 'sd-webui-controlnet',
                         'stable-diffusion-webui-images-browser', 'stable-diffusion-webui-rembg', 'SwinIR']
14:25:59-236841 INFO     Enabled extensions: ['canvas-zoom', 'deforum-for-automatic1111-webui', 'sd-dynamic-prompts',
                         'sd_dreambooth_extension', 'sd_webui_SAG', 'ultimate-upscale-for-automatic1111']
14:25:59-241345 INFO     Verifying requirements
**14:25:59-249352 WARNING  Package wrong version: accelerate 0.19.0 required 0.20.3**
14:25:59-250853 INFO     Installing package: accelerate==0.20.3
**14:26:01-410249 WARNING  Package wrong version: diffusers 0.16.1 required 0.17.1**
14:26:01-411750 INFO     Installing package: diffusers==0.17.1
**14:26:04-656824 WARNING  Package wrong version: transformers 4.29.2 required 4.26.1**
14:26:04-658826 INFO     Installing package: transformers==4.26.1
14:26:15-927161 INFO     Verifying packages
14:26:15-929163 INFO     Installing repositories
14:26:18-727509 INFO     Installing submodules
> 

Thanks :)

**Comment by brknsoul** [comment]
First, regarding the warnings, you'll notice that directly under them, the correct version is being downloaded; 
```
14:25:59-249352 WARNING Package wrong version: accelerate 0.19.0 required 0.20.3
14:25:59-250853 INFO Installing package: accelerate==0.20.3
```

Second, if you're using an nVidia card, you should enable xformers in Settings > Optimizations *(edit)* on slower systems. Otherwise use Scale-Dot-Product.

**Comment by vladmandic** [comment, marked as answer]
enabling xformers is optional and i'd only recommend it on slower systems. default is sdp and you don't need to enable anything. but if you want to use xformers, feel free to enable them, there are no downsides, just there are no upsides as well.

regarding packages wrong versions - if this is happening on every restart, that means that some extension keeps installing wrong versions. extensions should never change system dependencies, but some don't care and do it anyhow.

**Comment by gohan2091** [comment]
> First, regarding the warnings, you'll notice that directly under them, the correct version is being downloaded;
> 
> ```
> 14:25:59-249352 WARNING Package wrong version: accelerate 0.19.0 required 0.20.3
> 14:25:59-250853 INFO Installing package: accelerate==0.20.3
> ```
> 
> Second, if you're using an nVidia card, you should enable xformers in Settings > Optimizations _(edit)_ on slower systems. Otherwise use Scale-Dot-Product.

Yes, I am aware of your first point and I did say "When I close A1111 and reopen, the same warnings appear" but thank you. On your second point Vladmandic says after your post that there is no benefit to enable xformers for me so I guess I will leave them out. I can generate a 512x768 image in under 1 second so happy with the speed. Obviously things slow down when I perform batches. If there is anything I should have enabled to speed things up then of course I would like to ensure I have that turned on. What is Scale-Dot-Product?

> enabling xformers is optional and i'd only recommend it on slower systems. default is sdp and you don't need to enable anything. but if you want to use xformers, feel free to enable them, there are no downsides, just there are no upsides as well.
> 
> regarding packages wrong versions - if this is happening on every restart, that means that some extension keeps installing wrong versions. extensions should never change system dependencies, but some don't care and do it anyhow.

Thanks for the information about xformers, appreciated :)

I will take a look at my extensions but I am not running too much other than what came with your edition of A1111.

**Comment by vladmandic** [comment]
oversimplified, sdp is a newer alternative to xformers and its enabled by default in this app.

**Comment by gohan2091** [comment]
> oversimplified, sdp is a newer alternative to xformers and its enabled by default in this app.

Good to know. Appreciate your help and thanks for vlads A1111, it's really very good and excited to see how it improves further over time!

**Comment by gohan2091** [comment]
Discovered the troubled extension causing the 3 warnings was sd_dreambooth_extension. For a good 30 minutes though turning clip-interrogator-ext on and off sometimes gave the error too but seems to have not occured anymore. I have uninstalled the dreambooth extension.
